### DISCUSSION TITLE
Create Keycloak extension

### DISCUSSION THREAD

**Discussion by tiagogalvao7** [keycloak/keycloak#28180] [Adapter deprecation]
Hello guys, how can i develop a new Keycloak extension?

**Comment by codespearhead** [comment]
What you need is called [Keycloak SPI (Service Provider Interface) Framework](https://www.keycloak.org/docs/24.0.2/server_development/#_extensions). It might be worth looking into the codebase of some [Keycloak extensions](https://www.keycloak.org/extensions.html) in case you haven't already.
